### PR TITLE
Support crate keyword.

### PR DESCRIPTION
**Pull request by davidtwco**
In the 2018 edition, the `crate` keyword [is valid as a visibility modifier](https://rust-lang-nursery.github.io/edition-guide/2018/transitioning/modules/path-clarity.html). This PR modifies the syntax highlighting to reflect this (mostly because `rustc` uses it and seeing it mishighlighted has been bugging me).

### CODE REVIEW

File: syntax/rust.vim [hunk 1]
```diff
@@ -32,6 +32,7 @@ syn keyword   rustKeyword     fn nextgroup=rustFuncName skipwhite skipempty
 syn keyword   rustKeyword     in impl let
 syn keyword   rustKeyword     macro
 syn keyword   rustKeyword     pub nextgroup=rustPubScope skipwhite skipempty
+syn keyword   rustKeyword     crate skipwhite skipempty
```
Comment (chris-morgan) [inline syntax/rust.vim]: `skipwhite skipempty` is only needed (or desirable) if `nextgroup` is specified. Also note that all these keywords are in alphabetical order so far, so putting this one with or after `continue` (I have no particular preference) would be good.

File: syntax/rust.vim [hunk 2]
```diff
@@ -43,8 +44,6 @@ syn keyword   rustKeyword     mod trait nextgroup=rustIdentifier skipwhite skipe
 syn keyword   rustStorage     move mut ref static const
 syn match rustDefault /\<default\ze\_s\+\(impl\|fn\|type\|const\)\>/
 
-syn keyword   rustInvalidBareKeyword crate
-
```
Comment (chris-morgan) [inline syntax/rust.vim]: Remember to remove its `hi def link` line too.

File: syntax/rust.vim [hunk 3]
```diff
@@ -154,11 +153,11 @@ syn region    rustDerive      start="derive(" end=")" contained contains=rustDer
 syn keyword   rustDeriveTrait contained Clone Hash RustcEncodable RustcDecodable Encodable Decodable PartialEq Eq PartialOrd Ord Rand Show Debug Default FromPrimitive Send Sync Copy
 
 " dyn keyword: It's only a keyword when used inside a type expression, so
-" we make effort here to highlight it only when Rust identifiers follow it 
+" we make effort here to highlight it only when Rust identifiers follow it
```
Comment (chris-morgan) [inline syntax/rust.vim]: This and the one below it are not pertinent to the matter. Ideally, all the crate stuff will be in one commit, and this other stuff will either not be done, or be in a separate commit. Practice good commit hygiene! It’s fun! Sometimes it’s even useful afterwards! ☺

Comment (davidtwco) [reply syntax/rust.vim]: Oops, I didn't even notice this, the editor did it automatically. I'll separate it out.

File: syntax/rust.vim (hunk 2 above)
Comment (davidtwco) [reply syntax/rust.vim]: Should be fixed now.

File: syntax/rust.vim (hunk 1 above)
Comment (davidtwco) [reply syntax/rust.vim]: Should be fixed now.

File: syntax/rust.vim (hunk 3 above)
Comment (chris-morgan) [reply syntax/rust.vim]: A few years ago I got tired of that sort of thing happening, and so I started always using `--patch` on my `git add` invocations when modifying existing files (when adding new files you can’t), and verifying that the diff is in fact what I want to commit. With aliases, this means I now type `g ap`.

The staging area can definitely be helpful if you deliberately make it a part of your process.

Comment (da-x) [reply syntax/rust.vim]: Sorry, it was my editor that introduced those extraneous white spaces. I recently moved to vim-plug and forgot to re-enable `bronson/vim-trailing-whitespace`.
